pkg/instancetype/upgrade/handler.go

```diff
@@ -17,40 +16,52 @@
  * Copyright 2024 Red Hat, Inc.
  *
  */
-package instancetype
+package upgrade
 
 import (
 	"context"
 
 	appsv1 "k8s.io/api/apps/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
 
 	virtv1 "kubevirt.io/api/core/v1"
 	instancetypeapi "kubevirt.io/api/instancetype"
+	"kubevirt.io/client-go/kubecli"
 	"kubevirt.io/client-go/log"
 
 	"kubevirt.io/kubevirt/pkg/apimachinery/patch"
 	"kubevirt.io/kubevirt/pkg/instancetype/compatibility"
+	"kubevirt.io/kubevirt/pkg/instancetype/find"
+	"kubevirt.io/kubevirt/pkg/instancetype/revision"
 )
 
-type Upgrader interface {
-	Upgrade(vm *virtv1.VirtualMachine) error
+type Upgrader struct {
```

Comment:
When I see a public struct in a library, I always assume that it's allowed to create it directly in user code: 
```go
var u = &Upgrader{}
```
It would initialize private fields with zero values. 
For example, `sync.Mutex` is created this way.

But in this case, it would create an invalid struct. We need to use `New()` to create correctly initialized struct. 

In my opinion, it would be better to use an interface and a private struct to force users to use `New()`. But if you disagree, you don't have to change the code.

Reply:
@akrejcir yup that's a great point and something I think I missed in the original design proposal for these changes. I've posted https://github.com/kubevirt/kubevirt/pull/13036/commits/e7886fdc9404ce130c8de1f56cff1d7d45e0e754 to rework this if you could take a look.

Reply:
Excellent point!